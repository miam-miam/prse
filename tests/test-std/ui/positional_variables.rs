use prse::parse;

fn main() {
    let l = "zero one two";
    let num;

    parse!(l, "{0} {0} {1}");
    parse!(l, "{0} {num} {2}");
    parse!(l, "{0} {1} {}");
    parse!(l, "{2} one {0}");
}
