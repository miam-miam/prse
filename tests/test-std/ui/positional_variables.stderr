error: Each positional argument much uniquely map to a corresponding index in the returned tuple.
 --> ui/positional_variables.rs:7:15
  |
7 |     parse!(l, "{0} {0} {1}");
  |               ^^^^^^^^^^^^^

error: Each positional argument much uniquely map to a corresponding index in the returned tuple.
 --> ui/positional_variables.rs:8:15
  |
8 |     parse!(l, "{0} {num} {2}");
  |               ^^^^^^^^^^^^^^^

error: Cannot use implied positional arguments with explicitly defined ones.
 --> ui/positional_variables.rs:9:15
  |
9 |     parse!(l, "{0} {1} {}");
  |               ^^^^^^^^^^^^

error: Each positional argument much uniquely map to a corresponding index in the returned tuple.
  --> ui/positional_variables.rs:10:15
   |
10 |     parse!(l, "{2} one {0}");
   |               ^^^^^^^^^^^^^
